### ISSUE TITLE
Linux 6.4 memory corruption

### ISSUE THREAD

**Issue by fee1-dead** [NixOS/nixpkgs#242185]
### Describe the bug
According to [this Arch Linux bug](https://bugs.archlinux.org/task/78973), "per-vma locking during page fault handling" doesn't seem to work properly and can cause firefox to randomly crash.

### Steps To Reproduce
1. open firefox
2. firefox randomly crashes after being open for some time

### Expected behavior

firefox to not crash

### Additional context

I can confirm that on my system the relevant config options are set:

```
> cat /proc/config.gz | gzip -d | grep VMA_LOCK
CONFIG_ARCH_SUPPORTS_PER_VMA_LOCK=y
CONFIG_PER_VMA_LOCK=y
# CONFIG_PER_VMA_LOCK_STATS is not set
```

Not sure if this is the correct place to report a bug like this.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.4.0, NixOS, 23.11 (Tapir), 23.11.20230704.78419ed`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - channels(root): `"nixos"`
 - channels(beef): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by huantianad** [comment]
I have been noticing random crashes when using librewolf, often after clicking on a link and the page is just loading in. Unsure if it's the same issue, it does core dump due to "illegal hardware instruction" and occurred on 6.4.0 and 6.4.1

**Comment by fee1-dead** [comment]
@huantianad try going to about:crashes and check the crash reports. For me the path that crashed are pretty random (no pattern) with some only occurring on nixos systems

**Comment by huantianad** [comment]
For some reason, going to that tells me "Hmm. That address doesn’t look right. Please check that the URL is correct and try again." Not sure why, might be a librewolf thing, I can't find anything with a quick google search, but I'll check again in the morning.

**Comment by Panadestein** [comment]
Thanks for reporting this. I can also confirm this crashes, that look very random to me. I suspected that I was something memory related, because when the crash happens my system feels very sluggish, and sometimes I have to reboot. Furthermore, I tried using the beta Firefox, but it also happened there. From the `about:crashes` I could not find any hint on the cause of the problem. In the linked Bugzilla thread there is a [workaround](https://lore.kernel.org/all/20230703182150.2193578-1-surenb%40google.com/), hopefully it will be implemented soon.

**Comment by euank** [comment]
Those MM issues are supposed to be fixed in 6.4.3 https://github.com/NixOS/nixpkgs/pull/242785

It's already made its way to the unstable channel at least, so `nix flake update` / `nix-channel --update` and rebuilding should hopefully fix this.!

**Comment by euank** [comment]
I haven't had any crashes since updating my kernel to 6.4.3, and I was observing firefox crashes every few hours before that.

Optimistically closing this as resolved by the kernel update. Thanks all for figuring out the source of the crashes!
